driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -158,24 +158,51 @@ public Connection addConnectionToClose(int shardId, Connection c) {
     }
   }
 
-  private boolean canUseAdvancedShardAwareness() {
+  private boolean canUseAdvancedShardAwareness(boolean logResult) {
     ShardingInfo shardingInfo = host.getShardingInfo();
     if (shardingInfo == null) {
+      if (logResult) {
+        logger.warn(
+            "Not using advanced port-based shard awareness with "
+                + host
+                + " because sharding info is missing");
+      }
       return false;
     }
     if (!manager.configuration().getProtocolOptions().isUseAdvancedShardAwareness()) {
+      if (logResult) {
+        logger.warn(
+            "Not using advanced port-based shard awareness with "
+                + host
+                + " because it's disabled in configuration");
+      }
       return false;
     }
 
     boolean isSSLUsed = null != manager.configuration().getProtocolOptions().getSSLOptions();
     if (shardingInfo.getShardAwarePort(isSSLUsed) == 0) {
+      if (logResult) {
+        logger.warn(
+            "Not using advanced port-based shard awareness with "
+                + host
+                + " because we're missing port-based shard awareness port on the server");
+      }
       return false;
     }
 
     if (System.currentTimeMillis() < advShardAwarenessBlockedUntil) {
+      if (logResult) {
+        logger.warn(
+            "Not using advanced port-based shard awareness with "
+                + host
+                + " because of a previous error");
+      }
       return false;
     }
 
+    if (logResult) {
+      logger.info("Using advanced port-based shard awareness with " + host);
```

Comment:
```suggestion
      logger.info("Using advanced port-based shard awareness with {}", host);
```

(similar fix also in other `logger` calls)